packages/google_adsense/lib/src/ad_unit_widget_web.dart

```diff
@@ -0,0 +1,201 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:developer';
+import 'dart:js_interop';
+import 'dart:js_interop_unsafe';
+
+import 'package:flutter/widgets.dart';
+import 'package:web/web.dart' as web;
+
+import '../google_adsense.dart';
+import 'js_interop/adsbygoogle.dart';
+
+/// Widget displaying an ad unit
+class AdUnitWidgetWeb extends AdUnitWidget {
+  // TODO(sokoloff06): consider builder?
+  /// Constructs [AdUnitWidgetWeb]
+  AdUnitWidgetWeb({
+    required String adClient,
+    required String adSlot,
+    required bool isAdTest,
+    required Map<String, dynamic> additionalParams,
+    String? cssText,
+    super.key,
+  })  : _adClient = adClient,
+        _adSlot = adSlot,
+        _isAdTest = isAdTest,
+        _additionalParams = additionalParams {
+    _insElement
+      ..className = 'adsbygoogle'
+      ..style.display = 'block';
+    if (cssText != null && cssText.isNotEmpty) {
+      _insElement.style.cssText = cssText;
+    }
+    final Map<String, String> dataAttrs =
+        Map<String, String>.of(<String, String>{
+      AdUnitParams.AD_CLIENT: 'ca-pub-$_adClient',
+      'adSlot': _adSlot,
+    });
+    if (_isAdTest) {
+      dataAttrs.addAll(<String, String>{_AD_TEST_KEY: 'on'});
+    }
+    for (final String key in dataAttrs.keys) {
+      _insElement.dataset
+          .setProperty(key as JSString, dataAttrs[key]! as JSString);
+    }
+    if (_additionalParams.isNotEmpty) {
+      for (final String key in _additionalParams.keys) {
+        _insElement.dataset.setProperty(
+            key as JSString, _additionalParams[key].toString() as JSString);
+      }
+    }
+  }
+
+  static const String _AD_TEST_KEY = 'adtest';
+
+  @override
+  String get adClient => _adClient;
+  final String _adClient;
+
+  @override
+  String get adSlot => _adSlot;
+  final String _adSlot;
+
+  @override
+  bool get isAdTest => _isAdTest;
+  final bool _isAdTest;
+
+  @override
+  Map<String, dynamic> get additionalParams => _additionalParams;
+  final Map<String, dynamic> _additionalParams;
+
+  final web.HTMLElement _insElement =
+      web.document.createElement('ins') as web.HTMLElement;
+
+  @override
+  State<AdUnitWidgetWeb> createState() => _AdUnitWidgetWebState();
+}
+
+class _AdUnitWidgetWebState extends State<AdUnitWidgetWeb>
+    with AutomaticKeepAliveClientMixin {
+  static int adUnitCounter = 0;
+  double adHeight = 1.0;
+  late web.HTMLElement adUnitDiv;
+
+  @override
+  bool get wantKeepAlive => true;
+
+  @override
+  Widget build(BuildContext context) {
+    super.build(context);
+    return SizedBox(
+      height: adHeight,
+      child: HtmlElementView.fromTagName(
+          tagName: 'div', onElementCreated: onElementCreated),
```

Comment:
Do we need a container `div` element, or could this be the `ins` directly?

Reply:
@ditman I tried to remove the wrapping div element but noticed that in such case some developer custom css rules can break ad behavior. 
E.g. When I applied a 5px border:

- Without wrapping div, right border is out of the viewport
![image](https://github.com/user-attachments/assets/8337a21f-c507-4ffe-83af-8e1bd76f1dc0)

- With wrapping div it is processed correctly:
![image](https://github.com/user-attachments/assets/1b63ba6b-e75e-4c94-baf2-ece0ab521f2c)

I don't know how important it is, I also played with paddins and margins and they caused overflows in both cases. I have a feeling there might be some conflict between Flutter platform view and adsense logic. 

Would love to hear your thoughts on this

Reply:
Agreeed to remove cssText property for now and document that Widget styling should be done on Flutter side

Reply:
@ditman I wll also keep the wrapping div for now I think as it make widget creation code a little more readable.